crates/cli/src/config.rs

```diff
@@ -889,6 +889,20 @@ Import an existing identity with:
         })
     }
 
+    pub fn can_set_name(&self, new_nickname: Option<&str>) -> Result<(), anyhow::Error> {
+        if let Some(new_nickname) = new_nickname {
+            if self.name_exists(new_nickname) {
+                return Err(anyhow::anyhow!("An identity with that name already exists."));
+            }
+
+            if is_hex_identity(new_nickname) {
+                return Err(anyhow::anyhow!("An identity name cannot be an identity."));
+            }
+        }
+
+        Ok(())
+    }
```

Comment:
how would you feel about making this like:
```rs
pub fn can_set_name(&self, new_nickname: &str) -> Result<(), anyhow::Error> {
```
(it seems a bit weird to call `can_set_name(None)`?)